src/EFCore.PG/Internal/NpgsqlLoggerExtensions.cs

```diff
@@ -48,40 +43,43 @@ public static class NpgsqlLoggerExtensions
         {
             var definition = NpgsqlResources.LogPrincipalTableNotInSelectionSet(diagnostics);
 
-            var warningBehavior = definition.GetLogBehavior(diagnostics);
-            if (warningBehavior != WarningBehavior.Ignore)
+            if (diagnostics.ShouldLog(definition))
             {
-                definition.Log(
-                    diagnostics,
-                    warningBehavior,
-                    foreignKeyName, tableName, principalTableName);
+                definition.Log(diagnostics, foreignKeyName, tableName, principalTableName);
             }
 
             // No DiagnosticsSource events because these are purely design-time messages
         }
 
-        /// <summary>
-        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
-        ///     directly from your code. This API may change or be removed in future releases.
-        /// </summary>
         public static void ColumnFound(
             [NotNull] this IDiagnosticsLogger<DbLoggerCategory.Scaffolding> diagnostics,
             [NotNull] string tableName,
             [NotNull] string columnName,
             [NotNull] string dataTypeName,
             bool nullable,
-            [CanBeNull] string defaultValue)
+            bool identity,
+            [CanBeNull] string defaultValue,
+            [CanBeNull] string computedValue)
         {
             var definition = NpgsqlResources.LogFoundColumn(diagnostics);
 
-            var warningBehavior = definition.GetLogBehavior(diagnostics);
-            if (warningBehavior != WarningBehavior.Ignore)
+            if (diagnostics.ShouldLog(definition))
             {
                 definition.Log(
                     diagnostics,
-                    warningBehavior,
-                    tableName, columnName, dataTypeName, nullable, defaultValue);
+                    l => l.LogDebug(
+                        definition.EventId,
+                        null,
+                        definition.MessageFormat,
+                        tableName,
+                        columnName,
+                        dataTypeName,
+                        nullable,
+                        identity,
+                        defaultValue,
+                        computedValue));
             }
```

Comment:
In that case brackets looks fine due to a deep indentation.

Reply:
Agreed, that's the style I usually prefer (omit curlies only when it really is a single line).